index.js

```diff
@@ -42,107 +47,63 @@ class MiHumidifier {
         // Device info
         this.infoService
             .setCharacteristic(Characteristic.Manufacturer, 'Xiaomi')
-            .setCharacteristic(Characteristic.Model, config.model)
-            .setCharacteristic(Characteristic.SerialNumber, 'Undefined');
+            .setCharacteristic(Characteristic.Model, config.model);
 
         // Active
-        if (this.humidifier.powerGetName && this.humidifier.powerSetName) {
-            this.humidifierService
-                .getCharacteristic(Characteristic.Active)
-                .on('get', this.getActive.bind(this))
-                .on('set', this.setActive.bind(this));
-        }
+        this.registerCharacteristic(CharacteristicOperation.GET, this.humidifierService, Characteristic.Active, this.humidifier.powerGetName, this.getActive);
```

Comment:
Here and below: `registerCharacteristic` requires 6 arguments, but only 5 passed